helm/prometheus-rules/templates/kaas/phoenix/alerting-rules/capa.inhibition.rules.yml

```diff
@@ -0,0 +1,34 @@
+{{- if eq .Values.managementCluster.provider.kind "capa" }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  creationTimestamp: null
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+    cluster_type: "management_cluster"
+  name: capa.inhibitions.rules
+  namespace: {{ .Values.namespace  }}
+spec:
+  groups:
+  - name: capa.inhibitions
+    rules:
+    - alert: InhibitionClusterWithoutWorkerNodes
+      annotations:
+        description: '{{`Cluster ({{ $labels.cluster_id }}) has no worker nodes.`}}'
+      expr: |-
+        label_replace(
+            capi_cluster_status_condition{type="ControlPlaneReady", status="True"},
+                "cluster_id",
+                "$1",
+                "name",
+                "(.*)"
+            ) == 1
+        unless on (cluster_id) (
+            sum(capi_machinepool_spec_replicas{} > 0) by (cluster_id)
```

Comment:
Do we accept 1 worker nodes only?

Reply:
Prometheus-agent/Alloy could run on a 1-node WC, so that's potentially ok.

Reply:
Are we sure we want to use `capi_machinepool_spec_replicas`? That's the number of replicas defined in the `MachinePool` spec, but that doesn't necessarily represent the _current_ number of nodes. For the inhibition, I thought we would prefer to use the current number of replicas. There are other metrics like

- `capi_machinepool_status_replicas`: Replicas is the most recently observed number of replicas.
- `capi_machinepool_status_replicas_ready`: The number of ready replicas for this MachinePool. A machine is considered ready when the node has been created and is "Ready".
- `capi_machinepool_status_replicas_available`: The number of available replicas (ready for at least minReadySeconds) for this MachinePool.

Do you think it would make sense to use one of those instead?

Reply:
Current inhibition works when the cluster has been purposely scaled down.

If the cluster should have nodes but none is ready/available, I think we should get a page.
I don't think current state of CAPI monitoring manages it, so I'd rather have a "prometheus-agent down" alert than no alert in this case.

It seems to me that there's quite a gap between vintage AWS and CAPA alerts, but I don't think it's Atlas responsibility to fix it. So I went the quickest way to solving my actual issue :sweat_smile:

Reply:
That would be ideal. But my first expectation is to not get paged when a cluster has no issues, for now :rofl: